### PR TITLE
Update the Besu EVM library to version 23.7.2

### LINKED ISSUE #8242: Bump Besu version
### Problem

A new version of Besu will be coming out soon.  Test with the new version once it is available

### Solution

Once a new version of Besu becomes available, update services to the latest version.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Use version 23.7.2 of the Besu EVM library

Fixes #8242

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/contracts/operation/HederaStaticCallOperationTest.java [hunk 1]
```diff
@@ -129,7 +129,7 @@ void executesAsExpected() {
         }
         given(evmMsgFrame.stackSize()).willReturn(20);
         given(evmMsgFrame.getRemainingGas()).willReturn(cost);
-        given(evmMsgFrame.getMessageStackDepth()).willReturn(1025);
+        given(evmMsgFrame.getDepth()).willReturn(1025);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/contracts/operation/HederaStaticCallOperationTest.java:132]: Jeez, what a gratuitous change ....

**Review by david-bakin-sl — APPROVED** [review]
LGTM!

File: hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmTxProcessor.java [hunk 2]
```diff
@@ -143,15 +139,14 @@ public HederaEvmTransactionProcessingResult execute(
...
-                .depth(0)
                 .completer(unused -> {})
                 .isStatic(isStatic)
                 .miningBeneficiary(coinbase)
                 .blockHashLookup(blockMetaSource::getBlockHash)
                 .contextVariables(Map.of("HederaFunctionality", getFunctionType()));
 
         this.initialFrame = buildInitialFrame(commonInitialFrame, receiver, payload, value);
-        messageFrameStack.addFirst(initialFrame);
+        var messageFrameStack = initialFrame.getMessageFrameStack();
```
Comment (qnswirlds) [inline hedera-node/hedera-evm/src/main/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmTxProcessor.java:149]: nit:
```suggestion
        final var messageFrameStack = initialFrame.getMessageFrameStack();
```

File: hedera-node/hedera-evm/src/test/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmTxProcessorTest.java [hunk 3]
```diff
@@ -295,31 +285,27 @@ void assertIsContractCallFunctionality() {
     @Test
     void assertTransactionSenderAndValue() {
         // setup:
-        doReturn(Optional.of(receiver)).when(transaction).getTo();
-        given(transaction.getSender()).willReturn(senderAddress);
-        given(transaction.getValue()).willReturn(Wei.of(1L));
+        Wei oneWei = Wei.of(1L);
```
Comment (qnswirlds) [inline hedera-node/hedera-evm/src/test/java/com/hedera/node/app/service/evm/contracts/execution/HederaEvmTxProcessorTest.java:288]: nit:
```suggestion
        final Wei oneWei = Wei.of(1L);
```

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/contracts/execution/CallEvmTxProcessorTest.java [hunk 4]
```diff
@@ -507,32 +498,28 @@ void assertIsContractCallFunctionality() {
     @Test
     void assertTransactionSenderAndValue() {
         // setup:
-        doReturn(Optional.of(receiver.getId().asEvmAddress())).when(transaction).getTo();
+        Wei oneWei = Wei.of(1L);
```
Comment (qnswirlds) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/contracts/execution/CallEvmTxProcessorTest.java:501]: nit:
```suggestion
        final Wei oneWei = Wei.of(1L);
```

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/contracts/execution/CallLocalEvmTxProcessorTest.java [hunk 5]
```diff
@@ -179,32 +172,28 @@ void assertIsContractCallFunctionality() {
     @Test
     void assertTransactionSenderAndValue() {
         // setup:
-        doReturn(Optional.of(receiver.getId().asEvmAddress())).when(transaction).getTo();
+        Wei oneWei = Wei.of(1L);
```
Comment (qnswirlds) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/contracts/execution/CallLocalEvmTxProcessorTest.java:175]: nit:
```suggestion
        final Wei oneWei = Wei.of(1L);
```

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/contracts/execution/CreateEvmTxProcessorTest.java [hunk 6]
```diff
@@ -248,31 +241,27 @@ void assertIsContractCallFunctionality() {
     @Test
     void assertTransactionSenderAndValue() {
         // setup:
-        doReturn(Optional.of(receiver.getId().asEvmAddress())).when(transaction).getTo();
-        given(transaction.getSender()).willReturn(sender.getId().asEvmAddress());
-        given(transaction.getValue()).willReturn(Wei.of(1L));
+        Wei oneWei = Wei.of(1L);
```
Comment (qnswirlds) [inline hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/contracts/execution/CreateEvmTxProcessorTest.java:244]: nit:
```suggestion
        final Wei oneWei = Wei.of(1L);
```

**Review by qnswirlds — DISMISSED** [review]
Nit: Missing a few `final` keywords.

**Review by tinker-michaelj — APPROVED** [review]
LGTM

**Review by david-bakin-sl — APPROVED** [review]
LGTM!

**Comment by Nana-EC** [comment]
@IvanKavaldzhiev can we open a ticket to update the mirror node Besu version to line up with this.
Should aim for 0.90.

**Comment by IvanKavaldzhiev** [comment]
> @IvanKavaldzhiev can we open a ticket to update the mirror node Besu version to line up with this. Should aim for 0.90.

We opened a [ticket](https://github.com/hashgraph/hedera-mirror-node/issues/6958).

**Pull request merged**